### DISCUSSION TITLE
Is there a way to dismiss the ShellRoute modal programmatically?

### DISCUSSION THREAD

**Discussion by bqubique** [fujidaiti/smooth_sheets#120] [Q&A]
I use `GoRouter` with `ShellRoute`'s as described in the examples. Inside the modal navigation I want to `pop` the whole modal sheets (the whole `ShellRoute` including sub-routes).
When creating the `NavigationSheet` I also tried looking into `SheetController` class, but it does not have any method to dismiss the page manually. 
This would be useful with a close button on any sheet.

Right now I've come up with this solution:
```
Navigator.of(context).popUntil((route) => route.isFirst);
Navigator.of(context).pop();
```

Which looks like an antipattern to using GoRouter solely. 
What can be done in this case?

**Comment by fujidaiti** [comment]
In go_router (and Navigator 2.0), you can go anywhere you want just by specifying the path to the location like in a website. For example, say we have the 3 pages; `/`: the home, `/a`: the first page of the shell route, which is stacked on top of the home (like the NavigationSheet in your case), and `/a/b`:  the second page of the shell route. Then, if you want to return to home from `/a/b`, simply call `context.go('/')`.

Adding the following app bar to any page in the NavigationSheet of the AI playlist generator example could be a good example of this mechanism.
```dart
appbar: AppBar(
  actions: [
    IconButton(
      // Returns to the home from any page in the sheet
      onPressed: () => context.go('/'),
      icon: const Icon(Icons.close),
    ),
  ],
),
```

**Reply by bqubique** [reply]
True, but what if the same route is being called in other routes as well? 
What if two different routes push this `ShellRoute`/Modal? How would I know where to `go` (besides passing a flag when calling this sheet)?

**Comment by fujidaiti** [comment]
>Right now I've come up with this solution:
>
>```dart
>Navigator.of(context).popUntil((route) => route.isFirst);
>Navigator.of(context).pop();
>```

Then I think this is the only solution as far as I understand it. 
If go_router package has an [API to obtain the current location](https://github.com/flutter/flutter/issues/129833) (e.g. `/a/b/c`), we might be able to extract the destination to `go` to, from the current location using `RegExp` or something.
